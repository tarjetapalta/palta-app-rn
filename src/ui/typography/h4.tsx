import React, { FunctionComponent } from 'react';

import { Text, TextProps } from './base';

export const H4: FunctionComponent<
  Pick<TextProps, 'testID' | 'color' | 'emphasis' | 'textAlign'>
> = ({
  /** Test id for test */
  testID,
  /** Text */
  children,
  /** Text color */
  color,
  /** Text font opacity */
  emphasis,
  /** Text align */
  textAlign,
}) => (
  <Text
    testID={testID}
    color={color}
    size="m"
    weight="regular"
    emphasis={emphasis}
    textAlign={textAlign}
  >
    {children}
  </Text>
);
