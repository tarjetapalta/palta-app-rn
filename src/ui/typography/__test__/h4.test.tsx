import React from 'react';
import renderer from 'react-test-renderer';

import WithTheme from '../../../theme/context';
import { H4 } from '../h4';

describe('h4 component', () => {
  it('basic H4 with color "text", emphasis "high", weight "medium" and testID "H4"', () => {
    const result = renderer
      .create(
        <WithTheme>
          <H4>Some text</H4>
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
  it('basic H4 with color "primary", emphasis "medium"', () => {
    const result = renderer
      .create(
        <WithTheme>
          <H4 color="primary" emphasis="medium">
            Some text
          </H4>
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
  it('basic with custom testId', () => {
    const result = renderer
      .create(
        <WithTheme>
          <H4 testID="custom testID">Some text</H4>
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
});
