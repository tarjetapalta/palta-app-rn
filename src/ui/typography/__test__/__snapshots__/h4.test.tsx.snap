// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`h4 component basic H4 with color "primary", emphasis "medium" 1`] = `
<Text
  color="primary"
  emphasis="medium"
  numberOfLines={0}
  size="m"
  style={
    Array [
      Object {
        "color": "#91b086",
        "fontFamily": "BwModelica-Regular",
        "fontSize": 16,
        "opacity": 0.6,
        "textAlign": "center",
      },
    ]
  }
  weight="regular"
>
  Some text
</Text>
`;

exports[`h4 component basic H4 with color "text", emphasis "high", weight "medium" and testID "H4" 1`] = `
<Text
  numberOfLines={0}
  size="m"
  style={
    Array [
      Object {
        "color": "#283238",
        "fontFamily": "BwModelica-Regular",
        "fontSize": 16,
        "opacity": 0.87,
        "textAlign": "center",
      },
    ]
  }
  weight="regular"
>
  Some text
</Text>
`;

exports[`h4 component basic with custom testId 1`] = `
<Text
  numberOfLines={0}
  size="m"
  style={
    Array [
      Object {
        "color": "#283238",
        "fontFamily": "BwModelica-Regular",
        "fontSize": 16,
        "opacity": 0.87,
        "textAlign": "center",
      },
    ]
  }
  testID="custom testID"
  weight="regular"
>
  Some text
</Text>
`;
